tests/Type/Symfony/data/cache.php

```diff
@@ -12,6 +12,15 @@ function testCacheCallable(\Symfony\Contracts\Cache\CacheInterface  $cache): voi
 	assertType('string', $result);
 };
 
+/**
+ * @param callable():string $fn
+ */
+function testNonScalarCacheCallable(\Symfony\Contracts\Cache\CacheInterface $cache, callable $fn): void {
+	$result = $cache->get('foo', $fn);
+
+	assertType('string', $result);
```

Comment:
I feel like `non-empty-string` should be generalized to `string`. Which means we'll likely have to always generalize by `GeneralizePrecision::templateArgument()`.

Reply:
wasn't aware `GeneralizePrecision::templateArgument()` is a thing 😅